zsl/interface/celery/worker.py

```diff
@@ -0,0 +1,101 @@
+"""
+:mod:`zsl.interface.celery.worker`
+----------------------------------
+
+Implementation of celery workers.
+
+.. moduleauthor:: Peter Morihladko
+"""
+from __future__ import unicode_literals
+
+import sys
+
+from celery import Celery, shared_task
+from injector import Module, singleton, provides
+
+from zsl.utils.injection_helper import inject
+from zsl.interface.task_queue import TaskQueueWorker
+from zsl.task.job_context import Job
+
+
+class CeleryTaskQueueWorkerBase(TaskQueueWorker):
+    """Base class for celery task queue worker.
+
+    It contains only the task execution part of worker.
+    """
+
+    def execute_celery_task(self, job_data):
+        # type: (dict) -> dict
+        """Creates job from task and executes the job.
+
+        :param job_data: job's data
+        :return: job results
+        :rtype: dict
+        """
+        job = Job(job_data)
+
+        return self.execute_job(job)
+
+
+class CeleryTaskQueueOutsideWorker(CeleryTaskQueueWorkerBase):
```

Comment:
Maybe we should remove the task execution logic from the worker, so this class would not have to exists. Worker would handle only the running and listening for tasks and the task execution could be done by a simple function or another interface. This would hit me in later stages of implementation so I kept it as it was in GearmanWorker.